### DISCUSSION TITLE
is possible to train Textcat with rules/matcher?

### DISCUSSION THREAD

**Discussion by info2000** [explosion/spaCy#9636] [Help: Coding & Implementations] [answered]
I want to use the matcher (phrase matcher) power using tokens and lemmas to build a textcat train dataset in front of a python regex that always will need exact word match
And also if this is the rigth way or is a better way to use rule 
on the dataset building process

```
matcher = PhraseMatcher(nlp.vocab, attr="LOWER")
patterns = [nlp.make_doc(name) for name in ["Angela Merkel", "Barack Obama"]]
matcher.add("Names", patterns)

doc = nlp("angela merkel and us president barack Obama")
doc.cats = categoriesText #from the csv in example
for match_id, start, end in matcher(doc):
    doc.cats["Names"]=1
    print("Matched based on lowercase token text:", doc.vocab.strings[match_id], doc[start:end])
```

Is this the  correct  way ? or exists a better way  on spacy to  augmentate a textcat dataset?

**Comment by polm** [comment, marked as answer]
It is definitely possible to use the Matcher to create training data for a textcat model. That's a form of "weak supervision", where you train a statistical model using the output of a rule-based model. 

The code you have works. I assume it's example code, but just in case, I will note that "names" is kind of a weird category for a document. Also note you can use entities from existing pipelines if you actually need to match on names. 

We recently released a [weak supervision tutorial project](https://github.com/explosion/projects/tree/576ba0089cdca320a9f56cde9f4220f12a1f6000/tutorials/ner_tweets) that you might find useful.
